src/ProtoStruct.jl

```diff
@@ -1,5 +1,27 @@
 
+const revise_uuid = Base.UUID("295af30f-e4ad-537b-8983-00126c2a3abe")
+const revise_pkgid = Base.PkgId(revise_uuid, "Revise")
+
+function checkrev()
+    revise_pkgid in keys(Base.loaded_modules) || return false 
+    d = @__DIR__ 
+    f = splitpath(@__FILE__)[end]
+    watched_files = getproperty(Base.loaded_modules[revise_pkgid], :watched_files)
+    d in keys(watched_files) || return false
+    trackedfiles = watched_files[d].trackedfiles
+    return f in keys(trackedfiles)
+end
+
 macro proto(expr)
+    if checkrev()
+        @eval __module__ $(_proto(expr))
```

Comment:
Can you explain to me what this is doing? Is it evaluating the struct definition in the ProtoStructs module?

Reply:
It will re-evaluate in the module the `proto` definition resides, so it should be okay